tests/rptest/tests/retention_policy_test.py

```diff
@@ -247,15 +247,18 @@ def test_shadow_indexing_default_local_retention(self,
             {"cloud_storage_enable_remote_write": cluster_remote_write},
             expect_restart=True)
 
+        config = {"cleanup.policy": TopicSpec.CLEANUP_DELETE}
+
+        # None case means use cluster default for topic creation.
+        if topic_remote_write is not None:
+            config |= {"redpanda.remote.write": topic_remote_write}
+
         self.rpk.create_topic(topic=self.topic_name,
                               partitions=1,
                               replicas=1,
-                              config={
-                                  "cleanup.policy": TopicSpec.CLEANUP_DELETE,
-                                  "redpanda.remote.write": topic_remote_write
-                              })
+                              config=config)
 
-        if topic_remote_write != "-1":
+        if topic_remote_write is not None:
```

Comment:
This test fails with the changes in this PR if the test changes are not made.

You are correct in seeing that `topic create topic -c redpanda.remote.write=-1` would be the original command executed in this test, which previously would succeed because we would hit the "silently ignore" case discussed in the cover letter. That was the author's intention, to use the cluster default, but this behavior is not supported in Kafka and therefore shouldn't be supported in `redpanda`.

Using `-1` as the value for a boolean returns an error in Kafka:

```
$ ./kafka-topics.sh --create --topic tapioca --bootstrap-server localhost:9092 --config preallocate=-1
Error while executing topic command : Invalid value -1 for configuration preallocate: Expected value to be either true or false
[2024-10-08 15:31:09,189] ERROR org.apache.kafka.common.config.ConfigException: Invalid value -1 for configuration preallocate: Expected value to be either true or false
	at org.apache.kafka.common.config.ConfigDef.parseType(ConfigDef.java:706)
	at org.apache.kafka.common.config.ConfigDef.parseValue(ConfigDef.java:531)
	at org.apache.kafka.common.config.ConfigDef.parse(ConfigDef.java:524)
	at org.apache.kafka.storage.internals.log.LogConfig.validate(LogConfig.java:692)
	at org.apache.kafka.storage.internals.log.LogConfig.validate(LogConfig.java:684)
	at org.apache.kafka.tools.TopicCommand.parseTopicConfigsToBeAdded(TopicCommand.java:176)
	at org.apache.kafka.tools.TopicCommand.access$000(TopicCommand.java:81)
	at org.apache.kafka.tools.TopicCommand$CommandTopicPartition.<init>(TopicCommand.java:267)
	at org.apache.kafka.tools.TopicCommand$TopicService.createTopic(TopicCommand.java:460)
	at org.apache.kafka.tools.TopicCommand.execute(TopicCommand.java:105)
	at org.apache.kafka.tools.TopicCommand.mainNoExit(TopicCommand.java:90)
	at org.apache.kafka.tools.TopicCommand.main(TopicCommand.java:85)
 (org.apache.kafka.tools.TopicCommand)
```

The test was written with this assumed behavior for `-1` meaning "cluster default" in mind, and they were accidentally correct.

I fixed the test here by just not setting a value for `redpanda.remote.write`, which achieves the same thing as originally intended.